test/unit/mongo_client.test.js

```diff
@@ -719,4 +720,59 @@ describe('MongoOptions', function () {
       expect(parse).to.throw(/URI/);
     });
   });
+
+  it('srvMaxHosts cannot be combined with LB or ReplicaSet', () => {
+    expect(() => {
+      new MongoClient('mongodb+srv://localhost?srvMaxHosts=2&replicaSet=repl');
+    }).to.throw(MongoParseError, 'Cannot use srvMaxHosts option with replicaSet');
+    expect(() => {
+      new MongoClient('mongodb+srv://localhost?srvMaxHosts=2&loadBalanced=true');
+    }).to.throw(MongoParseError, 'Cannot limit srv hosts with loadBalanced enabled');
+    expect(() => {
+      new MongoClient('mongodb+srv://localhost', { srvMaxHosts: 2, replicaSet: 'blah' });
+    }).to.throw(MongoParseError, 'Cannot use srvMaxHosts option with replicaSet');
+    expect(() => {
+      new MongoClient('mongodb+srv://localhost?loadBalanced=true', { srvMaxHosts: 2 });
+    }).to.throw(MongoParseError, 'Cannot limit srv hosts with loadBalanced enabled');
+  });
+
+  it('srvServiceName and srvMaxHosts cannot be used on a non-srv connection string', () => {
```

Comment:
we should add conditions for `srvMaxHosts` set to 0, since for non-srv we have to throw regardless of whether it's specifically a positive integer